### PR TITLE
fish: bump to 3.7.1

### PR DESCRIPTION
**Pull request by kenmays**
Built and tested on Haiku hrev58282 x86.
```
Unit tested: 93% tests passed, 18 tests failed out of 247
Total Test time (real) =  62.14 sec

The following tests FAILED:
          5 - env_vars (Failed)
         18 - convert (Failed)
         42 - fish_wcstod_underscores (Failed)
         48 - is_potential_path (Failed)
         82 - abbr.fish (Failed)
         88 - basic.fish (Failed)
         94 - cd.fish (Failed)
        119 - expansion.fish (Failed)
        136 - git.fish (Failed)
        153 - locale-numeric.fish (Failed)
        161 - path.fish (Failed)
        164 - printf.fish (Failed)
        176 - set.fish (Failed)
        181 - slices.fish (Failed)
        183 - status-command.fish (Failed)
        184 - status-value.fish (Failed)
        187 - string.fish (Failed)
        206 - vars_as_commands.fish (Failed
```

Closes: #11175
Closes:#1801

### CODE REVIEW

**Comment by Begasus** [comment]
No rocket needed when this hasn't been checked.

File: app-shells/fish/patches/fish-3.7.1.patchset [hunk 1]
```diff
@@ -0,0 +1,34 @@
+From bb8f751b7ef40a63f7b2eb2f6df9d5c257532f92 Mon Sep 17 00:00:00 2001
+From: Ken mays <kmays2000@gmail.com>
+Date: Tue, 22 Oct 2024 05:12:03 +0000
+Subject: Fix std::wcsto* issue for Haiku
```
Comment (korli) [inline app-shells/fish/patches/fish-3.7.1.patchset:4]: What's the issue actually? Is it reported in Haiku?

File: app-shells/fish/fish-3.7.1.recipe [hunk 2]
```diff
@@ -3,11 +3,12 @@ DESCRIPTION="Fish is a shell geared towards interactive use. Its features are \
...
 HOMEPAGE="https://ridiculousfish.com/shell/"
-COPYRIGHT="2005-2018 Axel Liljencrantz"
+COPYRIGHT="2005-2024 Axel Liljencrantz"
 LICENSE="GNU GPL v2"
-REVISION="2"
+REVISION="1"
 SOURCE_URI="https://github.com/fish-shell/fish-shell/releases/download/$portVersion/fish-$portVersion.tar.xz"
-CHECKSUM_SHA256="a6d45b3dc5a45dd31772e7f8dfdfecabc063986e8f67d60bd7ca60cc81db6928"
+CHECKSUM_SHA256="614c9f5643cd0799df391395fa6bbc3649427bb839722ce3b114d3bbc1a3b250"
+SOURCE_DIR="fish-$portVersion"
```
Comment (korli) [inline app-shells/fish/fish-3.7.1.recipe:11]: Why add SOURCE_DIR ?

File: app-shells/fish/patches/fish-3.7.1.patchset (hunk 1 above)
Comment (kenmays) [reply app-shells/fish/patches/fish-3.7.1.patchset:4]: https://dev.haiku-os.org/ticket/18207

Comment (kenmays) [reply app-shells/fish/patches/fish-3.7.1.patchset:4]: Duriing fish's 'configuration', spotted 'Looking for wcstod_l - not found'.... GCC13/Haiku x86.

File: app-shells/fish/fish-3.7.1.recipe [hunk 3]
```diff
@@ -25,7 +25,7 @@ PROVIDES="
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
-	cmd:gettext$secondaryArchSuffix
+	gettext${secondaryArchSuffix}
```
Comment (jmairboeck) [inline app-shells/fish/fish-3.7.1.recipe:28]: Why was this changed?

Comment (kenmays) [reply app-shells/fish/fish-3.7.1.recipe:28]: Didn't see a requirement for the command usage.

Comment (Begasus) [reply app-shells/fish/fish-3.7.1.recipe:28]: Did you check configure output? I'm pretty sure it is found in there.

Comment (kenmays) [reply app-shells/fish/fish-3.7.1.recipe:28]: Yes... it worked fine.

Comment (kenmays) [reply app-shells/fish/fish-3.7.1.recipe]: Let me be much more clear here my my response - thanks to begasus and OscarL:
'gettext - not found situation' - yes, I'm aware of it. Otherwise, seems to find what it needs (like: msgfmt) and builds without issue. Functionally, the shell worked for me - but will need more review time as I looking into the unit tests.

File: app-shells/fish/fish-3.7.1.recipe [hunk 4]
```diff
@@ -36,14 +35,15 @@ REQUIRES="
...
-	devel:libgettextlib$secondaryArchSuffix
 	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
 	devel:libncursesw$secondaryArchSuffix
 	devel:libpcre2_32$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
 	cmd:cmake
 	cmd:gcc$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
```
Comment (Begasus) [inline app-shells/fish/fish-3.7.1.recipe:46]: This is still wrong, it doesn't search for `cmd:gettext`

File: app-shells/fish/patches/fish-3.7.1.patchset (hunk 1 above)
Comment (korli) [reply app-shells/fish/patches/fish-3.7.1.patchset:4]: Please add the ticket URL in the patch message.
